packages/app/src/server/models/obsolete-page.js

```diff
@@ -498,16 +498,37 @@ export const getPageSchema = (crowi) => {
     await addConditionToFilteringByViewerForList(builder, user, showAnyoneKnowsLink);
 
     // count
-    const totalCount = await builder.query.exec('count');
+    // const totalCount = await builder.query.exec('count');
 
     // find
-    builder.addConditionToPagenate(opt.offset, opt.limit, sortOpt);
+    // builder.addConditionToPagenate(opt.offset, opt.limit, sortOpt);
+    const skip = opt.page ? (+opt.page - 1) * opt.limit : opt.offset;
+    const customLabels = {
+      totalDocs: 'totalCount',
+      docs: 'pages',
+      limit: 'limit',
+      page: 'currentPage',
+      nextPage: 'nextPage',
+      prevPage: 'prevPage',
+      totalPages: 'pageCount',
+    };
+
+    const paginationOptions = {
+      lean: true,
+      ...(opt.limit) && { limit: opt.limit },
+      ...(opt.offset) && { offset: skip },
+      ...(opt.page) && { page: opt.page },
+      sort: sortOpt,
+      customLabels,
+    };
+
     builder.populateDataToList(User.USER_FIELDS_EXCEPT_CONFIDENTIAL);
-    const pages = await builder.query.lean().clone().exec('find');
 
+    const paginatedPages = await Page.paginate(builder.query.clone(), paginationOptions);
     const result = {
-      pages, totalCount, offset: opt.offset, limit: opt.limit,
+      pages: paginatedPages.pages, totalCount: paginatedPages.totalCount, offset: opt.offset, limit: opt.limit,
     };
+
```

Comment:
findListFromBuilderAndViewer はいろんなところで使っているので改修するのは危険。
今回利用する API 以外に影響が出ない範囲で改修してください。

Reply:
The previous function has been reused, I have added a special condition for infinite scroll

Reply:
Don't support `opt.page`.
See my comment for `apiv3/pages.js`

Reply:
`opt.page` has been removed from models/obsolete-page.js

Reply:
以下の理由から、やはり `models/page.ts` に別のメソッドとして実装してほしい。

- このメソッドは findListByCreator, findListByCreator, findManageableListWithDescendants, findListWithDescendants などで使われているため、将来的にもし利用時に opt.offset の有無が変更されると挙動が大きく変わる
    - この改修は 影響範囲の広さを見積もれないので許容できない
    - ファイル名通り、「obsolete」なものとして手を入れないのが望ましい

Reply:
I want you to implement it as a new method at `models/page.ts` after all for the following reasons.

- This method is used by `findListByCreator`, `findManageableListWithDescendants`, `findListWithDescendants`, and so on. Therefore, the behavior will change significantly if the existence of `opt.offset` is updated when being used in the future.
    - This repair cannot be approved because we can't guess the impact on other parts
    - The filename `obsolete` suggests you should not edit it

Reply:
I have impelement new method in models/page.ts and revert the function back on obsolete-page to its original method